### PR TITLE
fix: strip whitespace from instance configuration values on save

### LINKED ISSUE #8737: 🐛 Bug: OAuth login fails when instance configuration values have trailing whitespace
## Description

When configuring OAuth credentials (e.g. GitHub Client ID) through the God Mode admin UI, any leading/trailing whitespace in the pasted value is persisted to the database as-is. This causes the OAuth redirect URL to include a URL-encoded space (`%20`) in the `client_id` parameter, which makes the identity provider return a 404.

This is easy to trigger — simply copy-pasting a Client ID from a web page often picks up a trailing space.

## Steps to reproduce

1. Self-host Plane v1.2.3 (also present on `preview` branch)
2. Go to God Mode → Authentication → GitHub → Configure
3. Paste a GitHub OAuth Client ID with a trailing space (e.g. `Ov23li2Dep2t79q18nxD `)
4. Save and enable GitHub SSO
5. Click "Login with GitHub" on the login page
6. The redirect URL contains `client_id=Ov23li2Dep2t79q18nxD%20`
7. GitHub returns a 404

## Expected behavior

Configuration values should be stripped of leading/trailing whitespace before being persisted to the database.

## Actual behavior

Whitespace is saved verbatim, breaking OAuth flows and potentially any other credential that is sensitive to extra characters.

## Root cause

`InstanceConfigurationEndpoint.patch()` in `apps/api/plane/license/api/views/configuration.py` saves the raw value from `request.data` without calling `.strip()`:

```python
value = request.data.get(configuration.key, configuration.value)
```

## Affected

All OAuth providers (GitHub, Google, GitLab, Gitea) and likely any credential field saved via the instance configuration endpoint.

## Environment

- Plane version: v1.2.3 (also present on current `preview` branch)
- Deployment: Self-hosted Docker Compose

### PR DESCRIPTION
**Pull request by rucoder**
## Summary

Fixes #8737

Trailing whitespace in OAuth credentials (e.g. `GITHUB_CLIENT_ID`) causes authentication failures. When a user pastes a value with leading/trailing spaces into the God Mode admin UI, the spaces are persisted to the database as-is. For OAuth client IDs this results in the identity provider rejecting the request — e.g. GitHub returns a 404 because the `client_id` parameter has a URL-encoded space (`%20`) appended.

## Change

Strip whitespace from all configuration values before persisting them in `InstanceConfigurationEndpoint.patch()`:

```python
# Before
value = request.data.get(configuration.key, configuration.value)

# After
value = str(
    request.data.get(configuration.key, configuration.value)
).strip()
```

## Steps to reproduce the bug

1. Self-host Plane v1.2.3
2. Go to God Mode → Authentication → GitHub → Configure
3. Paste a GitHub OAuth Client ID with a trailing space
4. Save and enable GitHub SSO
5. Click "Login with GitHub" on the login page
6. GitHub returns a 404 because `client_id` in the OAuth URL contains `%20`

## Testing

Verified manually on a self-hosted v1.2.3 instance. After applying this fix, the trailing space is stripped and OAuth redirects work correctly.

## Summary by CodeRabbit

* **Bug Fixes**
  * Configuration values are now automatically trimmed of leading and trailing whitespace when saved, preventing accidental extra spaces from being stored and ensuring null values remain null.

* **Tests**
  * Added unit tests covering whitespace trimming on saved configuration values: leading, trailing, both, already-clean values, and null preservation.

### CODE REVIEW

File: apps/api/plane/license/api/views/configuration.py [hunk 1]
```diff
@@ -45,7 +45,9 @@ def patch(self, request):
 
         bulk_configurations = []
         for configuration in configurations:
-            value = request.data.get(configuration.key, configuration.value)
+            value = str(
+                request.data.get(configuration.key, configuration.value)
+            ).strip()
```
[the remark this answers is not in the document]
Comment (rucoder) [reply apps/api/plane/license/api/views/configuration.py]: Good catch! Updated to use `isinstance(value, str)` guard instead of `str(...)` coercion, so `None` is preserved as-is. Also added a test case for the `null` scenario.
